### PR TITLE
MAINT: Update files as per pyansys standards

### PR DESCRIPTION
**Pull request by dipinknair**
- Update Contributing, Authors file
-  Replace `.flake8` with `pyproject.toml` file for pre-commit
- Remove `.codespelrc` since the ignored words are used from vocabularies
-  Update `pre-commit-config` with new and current changes
- change `codecov.yml` location
- Remove `.tox.ini` and move to `pyproject.toml`

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
-# Contributing
+# Contributors
 
-For contributing to this project, please refer to the [PyAnsys Developer's Guide].
+## Project Lead
 
-[PyAnsys Developer's Guide]: https://dev.docs.pyansys.com/
+* [Sam Sangili](https://github.com/samigithub2022)
```
Comment (RobPasMue) [inline CONTRIBUTING.md:5]: You might have to change this to also include @koubaa

File: pyproject.toml [hunk 2]
```diff
@@ -174,4 +180,78 @@ showcontent = true
...
+max-complexity = 10
+max-line-length = 100
+statistics = true
+
+# ========================================================
+# Tox Configuration
+# ========================================================
+
+[tool.tox]
+legacy_tox_ini = """
```
Comment (RobPasMue) [inline pyproject.toml:206]: Interesting, didn't know that you could do this!

**Review by RobPasMue — APPROVED** [review]
LGTM! Could you present this as well in tuesday's meeting? Especially flake8-pyproject and tox as part of the pyproject.toml config

**Comment by dipinknair** [comment]
> LGTM! Could you present this as well in tuesday's meeting? Especially flake8-pyproject and tox as part of the pyproject.toml config

Sure.

**Pull request merged**